### PR TITLE
[Storage] Support Soft Delete for file share

### PR DESCRIPTION
**Pull request by Juliehzl**
**Description**  

Add file share soft delete support
Customer can enable/disable share soft delete on a storage account File Service properties. After share soft delete is enable with a specific RetentionDays, The deleted Shared still can be list out with --include-deleted, and it can be restored to a normal share.

**Testing Guide**  

`az storage share-rm list` with `--include-deleted`
`az storage share-rm restore`

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S172

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 1]
```diff
@@ -1054,6 +1054,16 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
 
     with self.argument_context('storage share-rm list', resource_type=ResourceType.MGMT_STORAGE) as c:
         c.argument('account_name', storage_account_type, id_part=None)
+        c.argument('include_deleted', action='store_true',
+                   help='Include soft deleted file shares when specified.')
+
+    with self.argument_context('storage share-rm restore', resource_type=ResourceType.MGMT_STORAGE) as c:
+        c.argument('deleted_version',
+                   help='Identify the version of the deleted share that will be restored.')
+        c.argument('share_name',
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py:1063]: >share_name [](start = 20, length = 10)

Can we use share_name_type here too?

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 2]
```diff
@@ -1054,6 +1054,16 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
 
     with self.argument_context('storage share-rm list', resource_type=ResourceType.MGMT_STORAGE) as c:
         c.argument('account_name', storage_account_type, id_part=None)
+        c.argument('include_deleted', action='store_true',
+                   help='Include soft deleted file shares when specified.')
+
+    with self.argument_context('storage share-rm restore', resource_type=ResourceType.MGMT_STORAGE) as c:
+        c.argument('deleted_version',
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py:1061]: What's the type of deleted_version?

File: src/azure-cli/azure/cli/command_modules/storage/operations/file.py [hunk 3]
```diff
@@ -44,8 +44,25 @@ def get_stats(client, resource_group_name, account_name, share_name):
                       expand='stats')
 
 
-def update_share_rm(cmd, instance, metadata=None, share_quota=None, root_squash=None, access_tier=None):
+def list_share_rm(client, cmd, resource_group_name, account_name, include_deleted=None):
+
+    ListSharesExpand = cmd.get_models('ListSharesExpand', resource_type=ResourceType.MGMT_STORAGE)
+
+    expand = ListSharesExpand("deleted") if include_deleted is not None else None
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/storage/operations/file.py:51]: what's result of ListSharesExpand("deleted") if include_deleted is not None?

**Review by jsntcy — APPROVED** [review]
:shipit:

**Pull request merged**